bahmnicore-omod/src/main/resources/liquibase.xml

```diff
@@ -1252,38 +1277,28 @@
             update drug_order set dosing_type='org.openmrs.SimpleDosingInstructions' where dosing_type='SIMPLE';
         </sql>
     </changeSet>
-    <changeSet id="bahmni-core-201408130010" author="Mujir">
-        <createTable tableName="import_status" >
-            <column name="id" type="int" autoIncrement="true">
-                <constraints primaryKey="true" primaryKeyName="import_status_pk" nullable="false" />
-            </column>
-
-            <!-- Should the file name columns be text?? -->
-            <column name="original_file_name" type="varchar(500)">
-                <constraints nullable="false"/>
-            </column>
-            <column name="saved_file_name" type="varchar(500)" >
-                <!--<constraints nullable="false" unique="true"/>-->
-                <constraints nullable="false" />
-            </column>
-            <column name="error_file_name" type="varchar(500)"/>
-            <column name="type" type="varchar(25)">
-                <constraints nullable="false"/>
-            </column>
-            <column name="status" type="varchar(25)">
-                <constraints nullable="false"/>
-            </column>
-            <column name="successful_records" type="numeric(6)"/>
-            <column name="failed_records" type="numeric(6)"/>
-            <column name="stage_name" type="varchar(10)" />
-            <column name="stack_trace" type="text" />
-
-            <column name="uploaded_by" type="varchar(20)">
-                <constraints nullable="false"/>
-            </column>
-            <column name="start_time" type="timestamp"/>
-            <column name="end_time" type="timestamp"/>
-        </createTable>
+    <changeSet id="bahmni-core-201906170948" author="Mujir, Samuel34">
+    	<preConditions onFail="MARK_RAN">
+			<not><tableExists tableName="import_status"/></not>
+		</preConditions>
+    	<comment>Introducing import status table</comment>
+    	<sql>
+    		CREATE TABLE import_status (
```

Comment:
I see here you went to using a full SQL query to create the table, instead of using Liquibase XML fields. I don't know if one is better than the other, but was there a particular reason for you to do so?

Reply:
This was because of this error with mysql 
```Caused by: liquibase.exception.MigrationFailedException: Migration failed for change set liquibase.xml::bahmni-core-201906170948::Mujir, Samuel34:
     Reason: liquibase.exception.DatabaseException: Error executing SQL CREATE TABLE `import_status` (`id` INT AUTO_INCREMENT NOT NULL, `original_file_name` VARCHAR(500) NOT NULL, `saved_file_name` VARCHAR(500) NOT NULL, `error_file_name` VARCHAR(500) NULL, `type` VARCHAR(25) NOT NULL, `status` VARCHAR(25) NOT NULL, `successful_records` NUMERIC(6) NULL, `failed_records` NUMERIC(6) NULL, `stage_name` VARCHAR(10) NULL, `stack_trace` TEXT NULL, `uploaded_by` VARCHAR(20) NOT NULL, `start_time` TIMESTAMP, `end_time` TIMESTAMP, CONSTRAINT `import_status_pk` PRIMARY KEY (`id`)): Invalid default value for 'end_time':
          Caused By: Error executing SQL CREATE TABLE `import_status` (`id` INT AUTO_INCREMENT NOT NULL, `original_file_name` VARCHAR(500) NOT NULL, `saved_file_name` VARCHAR(500) NOT NULL, `error_file_name` VARCHAR(500) NULL, `type` VARCHAR(25) NOT NULL, `status` VARCHAR(25) NOT NULL, `successful_records` NUMERIC(6) NULL, `failed_records` NUMERIC(6) NULL, `stage_name` VARCHAR(10) NULL, `stack_trace` TEXT NULL, `uploaded_by` VARCHAR(20) NOT NULL, `start_time` TIMESTAMP, `end_time` TIMESTAMP, CONSTRAINT `import_status_pk` PRIMARY KEY (`id`)): Invalid default value for 'end_time':
          Caused By: Invalid default value for 'end_time'
```

Reply:
So liquibase xml attributes are not very flexible to handle this case that's why I chose to use SQL.

Reply:
you mean to say the existing XML definition of table does not work? I wonder how it worked before! maybe because of liquibase version change? I would prefer XML definition, instead of create table SQL. but if we can't find alt, then I am ok

Reply:
Yes it doesn't work to my surprise to! I'm however not sure why!